### PR TITLE
feat: claim campaign reward

### PR DESCRIPTION
**Pull request by nguyenhuudungz**

### CODE REVIEW

File: src/pages/Campaign/index.tsx [hunk 1]
```diff
@@ -215,6 +223,53 @@ export default function Campaign() {
     if (campaignsRefreshIn === 0 && selectedCampaign) mutate(SWR_KEYS.getLeaderboard(selectedCampaign.id))
   }, [mutate, campaignsRefreshIn, selectedCampaign])
 
+  const addTransactionWithType = useTransactionAdder()
+  const onClaimRewardSuccess = (response: TransactionResponse) => {
+    addTransactionWithType(response, {
+      type: 'Claim',
+      summary: 'campaign reward',
```
Comment (nguyenhuudungz) [inline src/pages/Campaign/index.tsx:230]: Update this.

**Review by tiendv89 — APPROVED** [review]

File: src/pages/Campaign/CampaignButtonWithOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    </StyledCampaignButtonWithOptions>
+  )
+}
+
+const StyledCampaignButtonWithOptions = styled(Button)`
+  position: relative;
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+    ${css`
+      flex: 1;
+    `}
```
Comment (namgold) [inline src/pages/Campaign/CampaignButtonWithOptions.tsx:161]: why so complex bro, didnt only `flex: 1;` solved it?

Comment (nguyenhuudungz) [reply src/pages/Campaign/CampaignButtonWithOptions.tsx:161]: My havit of wrapping it with css.

Comment (nguyenhuudungz) [reply src/pages/Campaign/CampaignButtonWithOptions.tsx:161]: It's good for linting.

Comment (namgold) [reply src/pages/Campaign/CampaignButtonWithOptions.tsx:161]: okay

File: package.json [hunk 3]
```diff
@@ -98,6 +98,7 @@
     "i18n:compile": "lingui compile",
     "i18n": "yarn i18n:extract && yarn i18n:compile",
     "start": "react-app-rewired start",
+    "start-local": "env-cmd -f .env.local react-app-rewired start",
```
Comment (namgold) [inline package.json:101]: we dont have `.env.local` file, do we? @nguyenhuudungz

File: src/pages/Campaign/CampaignButtonWithOptions.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+}) {
+  const theme = useTheme()
+  const [isShowNetworks, setIsShowNetworks] = useState(false)
+  const { changeNetwork } = useActiveNetwork()
+  const containerRef = useRef<HTMLButtonElement>(null)
+  useOnClickOutside(containerRef, () => setIsShowNetworks(false))
+  const { mixpanelHandler } = useMixpanel()
+
+  const chainIds: ChainId[] = campaign
+    ? campaign[type === 'enter_now' ? 'chainIds' : 'rewardChainIds'].split(',').map(item => +item)
```
Comment (namgold) [inline src/pages/Campaign/CampaignButtonWithOptions.tsx:41]: dont use `(item => +item)`. It's bad in semantic. Use `map(Number)`, it's better to understand and as shorter than this.

File: package.json (hunk 3 above)
Comment (nguyenhuudungz) [reply package.json:101]: I have .env.local file but don't push it. I like 50/50 when pushing this code.

File: src/pages/Campaign/CampaignButtonWithOptions.tsx (hunk 4 above)
Comment (nguyenhuudungz) [reply src/pages/Campaign/CampaignButtonWithOptions.tsx:41]: Ok will update this.

File: package.json (hunk 3 above)
Comment (namgold) [reply package.json:101]: Dont get your idea

Comment (namgold) [reply package.json:101]: What do you mean of `50/50 when pushing this code`

File: src/pages/Campaign/CampaignButtonWithOptions.tsx (hunk 4 above)
Comment (namgold) [reply src/pages/Campaign/CampaignButtonWithOptions.tsx:41]: Thanks

File: package.json (hunk 3 above)
Comment (nguyenhuudungz) [reply package.json:101]: When I dev backend, I have a lot of services in localhost. So creating .env.local is a convenient way for me to work around it. I think it doesn't bored you right.

Comment (namgold) [reply package.json:101]: okay bro, understand

**Pull request merged**